File: Easy/custom-game-of-life/custom-game-of-life.py
import sys
import math

h, w, n = [int(i) for i in input().split()]
alive = input()
dead = input()
cells = [list(input())for i in range(h)]

for _ in range(n):
    newCells = []
    for i in range(h):
        newRow = []
        for j in range(w):
            currentCell = cells[i][j]
            neighborCount = 0
            for k in range(max(0, i-1), min(h, i+2)):
                for l in range(max(0, j-1), min(w, j+2)):
                    neighborCount += cells[k][l] == "O"
            if currentCell == "O":
                neighborCount -= 1
                if alive[neighborCount]=="0":
                    currentCell = "."
            else:
                if dead[neighborCount]=="1":
                    currentCell = "O"
            newRow.append(currentCell)
        newCells.append(newRow)
    cells = newCells

for cell in cells:
    print("".join(cell))

File: Easy/custom-game-of-life/custom-game-of-life.js
var inputs = readline().split(' ');
const h = parseInt(inputs[0]);
const w = parseInt(inputs[1]);
const n = parseInt(inputs[2]);
const alive = readline();
const dead = readline();
let cells = [];
for (let i = 0; i < h; i++) {
    const line = readline();
    cells.push(line);
}

for (let x = 0; x < n; x++) {
    let newCells = [];
    for (let i = 0; i < h; i++) {
        let newRow = [];
        for (let j = 0; j < w; j++) {
            let currentCell = cells[i][j];
            let neighborCount = 0;
            for(let k = Math.max(0, i-1); k < Math.min(h, i+2); k++) {
                for(let l = Math.max(0, j-1); l < Math.min(w, j+2); l++) {
                    if (cells[k][l] === "O") neighborCount++;
                }
            }
            if (currentCell === "O") {
                neighborCount -= 1;
                if (alive[neighborCount] === "0")
                    currentCell = "."
            }
            else{
                if (dead[neighborCount] === "1")
                    currentCell = "O"
            }
            newRow.push(currentCell);
        }
        newCells.push(newRow);
    }
    cells = newCells;
}

for(cell of cells)
    console.log(cell.join(""));
